### PR TITLE
[Feature] Client Checksum Verification (Resubmit old 1678)

### PR DESCRIPTION
**Pull request by noudess**
This is a re submission of https://github.com/EQEmu/Server/pull/1678.

This is not my code, but it appeared from the original PR discussion that the issue was removing unrelated functionality.

I know at least Trust is using this PR.

If the rule is not enabled, it does not kick anyone, just supplies the framework to be added to.

I'm not looking to own this, I merely advanced the code that seemed to be agreed upon here, so some of us can use it without being custom, and someone with more knowledge can advance this as they see fit, beyond what this offers.

### CODE REVIEW

File: zone/command.cpp [hunk 1]
```diff
@@ -387,6 +387,7 @@ int command_init(void)
 		command_add("unscribespells", "- Clear out your or your player target's spell book.", AccountStatus::GMCoder, command_unscribespells) ||
 		command_add("untraindisc", "[spellid] - Untrain specified discipline from your target.", AccountStatus::GMCoder, command_untraindisc) ||
 		command_add("untraindiscs", "- Untrains all disciplines from your target.", AccountStatus::GMCoder, command_untraindiscs) ||
+		command_add("updatechecksum", "gm only", 250, command_updatechecksum) ||
```
Comment (fryguy503) [inline zone/command.cpp:390]: May want to use the AccountStatus:: Function instead of Magic Number here

Comment (noudess) [reply zone/command.cpp]: Done.

**Review by fryguy503 — APPROVED** [review]
LGTM

File: zone/CMakeLists.txt [hunk 2]
```diff
@@ -548,6 +548,7 @@ SET(gm_commands
     gm_commands/unmemspells.cpp
     gm_commands/unscribespell.cpp
     gm_commands/unscribespells.cpp
+	gm_commands/updatechecksum.cpp
```
Comment (Kinglykrab) [inline zone/CMakeLists.txt:551]: Formatting, this file is weird, think it's spaces.

File: zone/command.cpp [hunk 3]
```diff
@@ -387,6 +387,7 @@ int command_init(void)
 		command_add("unscribespells", "- Clear out your or your player target's spell book.", AccountStatus::GMCoder, command_unscribespells) ||
 		command_add("untraindisc", "[spellid] - Untrain specified discipline from your target.", AccountStatus::GMCoder, command_untraindisc) ||
 		command_add("untraindiscs", "- Untrains all disciplines from your target.", AccountStatus::GMCoder, command_untraindiscs) ||
+		command_add("updatechecksum", "gm only", AccountStatus::GMImpossible, command_updatechecksum) ||
```
Comment (Kinglykrab) [inline zone/command.cpp:390]: Could use a bit more verbosity here on the command note.

File: zone/gm_commands/updatechecksum.cpp [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+		if (c)
+		{
+			auto pack = new ServerPacket(ServerOP_ReloadRulesWorld, 0);
+			worldserver.SendPacket(pack);
+			c->Message(Chat::Red, "Successfully sent the packet to world to reload rules. (only world)");
+			safe_delete(pack);
+		}
+	}
+}
```
Comment (Kinglykrab) [inline zone/gm_commands/updatechecksum.cpp:21]: Is there a reason to check validity of `c` twice here?

**Review by Kinglykrab — APPROVED** [review]

File: zone/CMakeLists.txt (hunk 2 above)
Comment (noudess) [reply zone/CMakeLists.txt]: Fixed.

File: zone/gm_commands/updatechecksum.cpp (hunk 4 above)
Comment (noudess) [reply zone/gm_commands/updatechecksum.cpp:18]: No, this PR is tested and not my code so I was trying to not introduce any new issues by changing it.  That said, this extra check is clearly useless.  Fixed and ty.

File: zone/command.cpp (hunk 3 above)
Comment (noudess) [reply zone/command.cpp]: Yeah I didn't look at this when Im resurrected it - made it more descriptive.

**Comment by Akkadius** [comment]
I'll take a look at this one in a bit

**Comment by Akkadius** [comment]
This is still on my list. Haven't had an abundance of time. Will follow up soon.

**Comment by Akkadius** [comment]
Taking a look now, rewriting a bit of this before it gets merged

**Comment by Akkadius** [comment]
Made a few modifications

* Added log category `ChecksumVerification`
* Simplified field names
* Add more information to log messages and simplified log messages
* #updatechecksum now also reloads `variables` table in world, not just rules
* Tested both GM and non-GM failing and passing validation checks

```
[World] [ChecksumVerification] BaseData.txt validation [Passed] client [accountname] (1) has [269898431146477] status [255]
```

**Pull request merged**